### PR TITLE
Ltac2: put backtraces on more exceptions 

### LINKED ISSUE #16410: Ltac2 stack traces
#### Description of the problem

I'd like to have a global setting that enables stack trace printing for Ltac2 errors.  There should also be some support within the language for managing stack traces; in particular the `exn` type should be enriched with an optional stack trace that can then be manipulated in simple ways ("reraise the error with the explicit stack trace", "raise an error prepending the current frame to the stack", "raise an error starting the stack trace from scratch here"), and when stack-trace printing is disabled the stack-traces are always `None`.  This should not (I hope) add too much overhead (and I'm fine if it's the case that when we eventually get precompilation for Ltac2, changing this setting recompiles all Ltac2 tactics in scope).  But stack trace printing is essential for debugging.  (Having this will also pave the way for Ltac2 profiling, I believe.)

#### Coq Version

8.15

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #16410 (there may still be some that don't get caught)

### CODE REVIEW

File: plugins/ltac/tactic_debug.ml [hunk 1]
```diff
@@ -751,36 +751,21 @@ let skip_extensions trace =
...
-           if Option.is_empty best_loc ||
-              not (Option.is_empty loc) && Loc.finer loc best_loc
-           then
-             aux loc tail
-           else
-             aux best_loc tail
-        | [] -> best_loc in
-        aux loc trace in
-    best_loc, mt ()
+    mt ()
```
Comment (JasonGross) [inline plugins/ltac/tactic_debug.ml:763]: What's the reason for this removal?  It seems like it'll make Ltac error messages less informative?

Comment (SkySkimmer) [reply plugins/ltac/tactic_debug.ml:763]: The loc it produced was just thrown away.

**Comment by JasonGross** [comment]
This fails to build, but it would be amazingly useful for the debugging that I'm currently working on.  Any chance you could get it building soon?  (Ltac2 backtraces on anomalies are absolutely essential for being able to debug high performance Ltac2, since Constr.Unsafe gives no guarantees about lack of anomalies.)

**Comment by SkySkimmer** [comment]
Not sure how I missed that it didn't build x_x

**Review by ppedrot — APPROVED** [review]
As discussed offline, not the cleanest way to implement this but at least the PR is already written.

**Comment by JasonGross** [comment]
Now it builds, and has already proven very useful, thanks!

**Comment by ppedrot** [comment]
@SkySkimmer this needs a rebase now.

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**